### PR TITLE
Remove some default token stuff, add support for custom fixes

### PR DESCRIPTION
**Pull request by PatrickMcGrath29**
Removed some default fixes that were harder to toggle (like f string conversions and `yield from`), and then added a kwarg that will blacklist certain fixes / plugins from being run. It does that by filtering out the relevant plugin files when importing modules, so that they don't get run.

```
# Install the local version of pyupgrade
python setup.py install

pyupgrade <FILE> --py37-plus --fixes-to-exclude <FIX NAME 1> <FIX NAME 2...>
```

A fix name is equivalent to the name of a file in the `_plugins` directory. The majority of the fixes have their own file. Some don't, and for those it was harder to dynamically toggle them using the kwarg (it's definitely possible though), and so I just deleted that code if I didn't think it should be included as a default.

### CODE REVIEW

File: pyupgrade/_plugins/legacy.py [hunk 1]
```diff
@@ -28,14 +28,6 @@
 FUNC_TYPES = (ast.Lambda, ast.FunctionDef, ast.AsyncFunctionDef)
 
 
-def _fix_yield(i: int, tokens: List[Token]) -> None:
```
Comment (BenBrostoff) [inline pyupgrade/_plugins/legacy.py:31]: Gotcha, so `yield_from` and `_to_fstring` are completely gone as you said in description.

File: pyupgrade/_main.py [hunk 2]
```diff
@@ -483,29 +397,13 @@ def _fix_tokens(contents_text: str, min_version: Version) -> str:
...
-            tokens[i] = token._replace(src=_fix_long(_fix_octal(token.src)))
-        elif token.name == 'STRING':
+        if token.name == 'STRING':
             tokens[i] = _fix_ur_literals(tokens[i])
             if remove_u:
                 tokens[i] = _remove_u_prefix(tokens[i])
             tokens[i] = _fix_escape_sequences(tokens[i])
         elif token.src == '(':
             _fix_extraneous_parens(tokens, i)
-        elif token.src == 'format' and i > 0 and tokens[i - 1].src == '.':
```
Comment (BenBrostoff) [inline pyupgrade/_main.py:495]: So walking through these to see if I have them right:
- Remove format for f-strings
- Remove `encode` in favor of byte literals
- Not sure on last one - something on newline tokens?

**Review by BenBrostoff — APPROVED** [review]
💯 , this is a nice touch for invalid choices:

```
pyupgrade: error: argument --fixes-to-exclude: invalid choice: 'a' (choose from 'default_encoding', 'dict_literals', 'generator_expressions_pep289', 'identity_equality', 'io_open', 'legacy', 'lru_cache', 'metaclass_type', 'mock', 'native_literals', 'new_style_classes', 'open_mode', 'oserror_aliases', 'pep584', 'percent_format', 'set_literals', 'six_base_classes', 'six_calls', 'six_metaclasses', 'six_remove_decorators', 'six_simple', 'subprocess_run', 'typing_pep563', 'typing_pep585', 'typing_pep604', 'typing_text', 'unpack_list_comprehension', 'versioned_branches')
```

My only thought is the rationale for excluding `encode` to bytes unless I'm missing something here.

File: pyupgrade/_main.py (hunk 2 above)
Comment (PatrickMcGrath29) [reply pyupgrade/_main.py:495]: Hey Ben, I ended up adding most of this back and putting the things that were questionable behind a fix flag, so they can be manually excluded

**Pull request merged**